From 77bdcfa7a454262b804f6951c825dec1ac430e14 Mon Sep 17 00:00:00 2001
From: Calum Lind <calumlind+deluge@gmail.com>
Date: Fri, 17 May 2013 15:14:55 +0000
Subject: Fix seeing double torrents in classic mode

Need to ignore torrent added events from state as they are already
loaded by _on_session_state
---
diff --git a/deluge/ui/gtkui/torrentview.py b/deluge/ui/gtkui/torrentview.py
index cb63142..c1a6ae4 100644
--- a/deluge/ui/gtkui/torrentview.py
+++ b/deluge/ui/gtkui/torrentview.py
@@ -738,8 +738,9 @@ class TorrentView(listview.ListView, component.Component):
         self.save_state()
 
     def on_torrentadded_event(self, torrent_id, from_state):
-        self.add_row(torrent_id)
-        self.mark_dirty(torrent_id)
+        if not from_state:
+            self.add_row(torrent_id)
+            self.mark_dirty(torrent_id)
 
     def on_torrentremoved_event(self, torrent_id):
         self.remove_row(torrent_id)
--
cgit v0.9.0.2
